editoast/src/views/v2/scenario.rs

```diff
@@ -0,0 +1,629 @@
+use crate::decl_paginated_response;
+use crate::error::Result;
+use crate::models::train_schedule::LightTrainSchedule;
+use crate::models::{Infra, List};
+use crate::modelsv2::scenario::{ScenarioModel, ScenarioWithDetailsModel};
+use crate::modelsv2::timetable::Timetable;
+use crate::modelsv2::{Changeset, Create, Model, Project, Retrieve, Study, Tags};
+use thiserror::Error;
+
+use crate::views::pagination::{PaginatedResponse, PaginationQueryParam};
+use crate::views::projects::{ProjectIdParam, QueryParams};
+use crate::views::scenario::{check_project_study, check_project_study_conn, ScenarioIdParam};
+use crate::views::study::StudyIdParam;
+use crate::DbPool;
+use actix_web::web::Query;
+use actix_web::{delete, HttpResponse};
+use chrono::{NaiveDateTime, Utc};
+
+use actix_web::get;
+use actix_web::patch;
+use actix_web::{
+    post,
+    web::{Data, Json, Path},
+};
+use derivative::Derivative;
+use editoast_derive::EditoastError;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+crate::routes! {
+    "/v2/projects/{project_id}/studies/{study_id}/scenarios" => {
+        create,
+        list,
+        "/{scenario_id}" => {
+            get,
+            delete,
+            patch,
+        }
+    }
+}
+
+crate::schemas! {
+    ScenarioCreateForm,
+    ScenarioPatchFormV2,
+    ScenarioV2,
+    ScenarioWithDetailsV2,
+    ScenarioResponseV2,
+    PaginatedResponseOfScenarioWithDetails,
+    LightTrainSchedule, // TODO: remove from here once train schedule is migrated
+}
+
+#[derive(Debug, Deserialize, Serialize, ToSchema)]
+pub struct ScenarioWithDetailsV2 {
+    #[serde(flatten)]
+    pub scenario: ScenarioV2,
+    pub infra_name: String,
+    pub trains_count: i64,
+}
+
+#[derive(Debug, Deserialize, Serialize, ToSchema)]
+pub struct ScenarioV2 {
```

Comment:
We don't need this struct since it's identical to `modelsv2::Scenario`. I'd just derive Serialize and ToSchema for Scenario and drop the From.